go/tools/gazelle/merger/merger.go

```diff
@@ -47,26 +40,30 @@ var (
 )
 
 // MergeWithExisting merges genFile with an existing build file at
-// existingFilePath and returns the merged file.
-func MergeWithExisting(genFile *bzl.File, existingFilePath string) (*bzl.File, error) {
+// existingFilePath and returns the merged file. If a "# gazelle:ignore" comment
+// is found in the file, nil will be returned. If an error occurs, it will be
+// logged, and nil will be returned.
+func MergeWithExisting(genFile *bzl.File, existingFilePath string) *bzl.File {
 	oldData, err := ioutil.ReadFile(existingFilePath)
 	if err != nil {
-		return nil, err
+		log.Print(err)
+		return nil
 	}
 	oldFile, err := bzl.Parse(existingFilePath, oldData)
 	if err != nil {
-		return nil, err
+		log.Print(err)
+		return nil
 	}
-	if err := shouldIgnore(oldFile); err != nil {
-		return nil, err
+	if shouldIgnore(oldFile) {
+		return nil
 	}
 
 	oldStmt := oldFile.Stmt
 	var newStmt []bzl.Expr
 	for _, s := range genFile.Stmt {
 		genRule, ok := s.(*bzl.CallExpr)
 		if !ok {
-			return nil, fmt.Errorf("got %v expected only CallExpr in %q", s, genFile.Path)
+			log.Panicf("got %v expected only CallExpr in %q", s, genFile.Path)
```

Comment:
This error should only happen due to a bug in `rules` or `packages`. It should not be triggered by user input. I'd like to use `Fatal` for unrecoverable user errors (e.g., invalid flags on the command line) and `panic` for programmer errors so we get a stack.